### PR TITLE
Linter: update prospector

### PR DESCRIPTION
**Pull request by stsewd**
Requires https://github.com/readthedocs/common/pull/94

### CODE REVIEW

**Comment by stsewd** [comment]
There are a lot of similar warnings to fix... I'm going to skip updating prospector here and just submit a subset of the fixes before this grows too large :D

File: readthedocs/api/v2/views/integrations.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def post(self, request, project_slug):
                 resp = {'detail': 'This project is currently disabled'}
                 return Response(resp, status=status.HTTP_406_NOT_ACCEPTABLE)
         except Project.DoesNotExist:
-            raise NotFound('Project not found')
+            raise NotFound('Project not found') from None
```
Comment (stsewd) [inline readthedocs/api/v2/views/integrations.py:86]: probably I'll ignore these type of warnings, as it's just an esthetical change in the traceback.

File: readthedocs/builds/forms.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def clean_match_arg(self):
         except Exception:
             raise forms.ValidationError(
                 _('Invalid Python regular expression.'),
-            )
+            ) from None
```
Comment (humitos) [inline readthedocs/builds/forms.py:176]: If we are going to use `from None` on these, shouldn't we use the context? Like:

```python
        except Exception as e:
            raise forms.ValidationError(
                _('Invalid Python regular expression.'),
            ) from e
```

Comment (stsewd) [reply readthedocs/builds/forms.py:176]: I'm just going to remove this rule

**Comment by humitos** [comment]
What's the work missing here? Would you like to update the PR with those changes required or should we close it?

I think we are moving away from linting tho, since we are implementing black / darker in https://github.com/readthedocs/readthedocs.org/pull/8875. So, it makes more sense to close this PR probably. Feel free to re-open if you think it's better.

**Pull request closed without merging**